cpp/include/raft/random/multi_variable_gaussian.cuh

```diff
@@ -59,6 +59,81 @@ class multi_variable_gaussian : public detail::multi_variable_gaussian_impl<T> {
   ~multi_variable_gaussian() { deinit(); }
 };  // end of multi_variable_gaussian
 
+/**
+ * @brief Matrix decomposition method for `compute_multi_variable_gaussian` to use.
+ *
+ * `compute_multi_variable_gaussian` can use any of the following methods.
+ *
+ * - `CHOLESKY`: Cholesky decomposition
+ * - `JACOBI`:
+
+ */
+using detail::multi_variable_gaussian_decomposition_method;
+using detail::multi_variable_gaussian_setup_token;
+
+template <typename ValueType>
+multi_variable_gaussian_setup_token<ValueType> setup_multi_variable_gaussian(
+  const raft::handle_t& handle,
```

Comment:
I like the overloads here- this should make it much easier to pull it from the handle.

Reply:
@cjnolet I've actually removed the public "setup" methods.  They should no longer be needed, since we now take a memory resource for the workspace.